### ISSUE TITLE
.net core support

### ISSUE THREAD

**Issue by tameraw** [Azure/azure-iot-sdk-csharp#31]
_From @ondrejtomcik on October 22, 2016 20:44_

Hello,
we're using .net core and I am missing support for Microsoft.Azure.Devices . When it will be added please?
Service fabric already supports .net core but this main IoT SDK, which should be first in my opinion, is still missing.

Thanks

_Copied from original issue: Azure/azure-iot-sdks#887_

**Comment by tameraw** [comment]
_From @olivierbloch on October 25, 2016 17:25_

Hi @ondrejtomcik 
We do have this in our backlog but haven't prioritized over other features yet and we cannot commit to an ETA just yet either.
We understand the importance of supporting .Net Core, and if you want to give it a shot before we can get to it, we'd be more than happy to accept your contribution and can even try and point  you in the right direction to make this happen.
We'll update the thread as soon as we have a clearer idea of when the work will be done.
Thanks

**Comment by tameraw** [comment]
_From @ondrejtomcik on October 26, 2016 6:11_

Okay, thanks for the info. I will try to find some time for contribution. Can we discuss more deeply what should be done to have at least the idea where to begin?

**Comment by tameraw** [comment]
_From @arturl on October 27, 2016 4:6_

Hi @ondrejtomcik,

I suggest looking at the work we did to add support for UWP. Create a new project of type "Class Library (.NET Core)", start moving the source files from the Microsoft.Azure.Devices.csproj there, and see what breaks. You will discover a few APIs not supported for .NET Core, which will be a similar set to UWP -- look at how WINDOWS_UWP was used in the code.

If you succeed in pulling it off, I recommend creating a shared project to minimize duplication in multiple csproj files, as we did [here](https://github.com/Azure/azure-iot-sdks/tree/master/csharp/device/Microsoft.Azure.Devices.Client.Shared).

Good luck, and thanks so much for your help!

**Comment by tameraw** [comment]
_From @ondrejtomcik on October 27, 2016 6:13_

@arturl thank you. I'll try to find a time to make it happen.

**Comment by tameraw** [comment]
_From @ondrejtomcik on October 27, 2016 14:38_

Hello again @arturl @olivierbloch,
I would like to ask you if you know when https://github.com/Azure/azure-amqp will be released? Repo updated to work with .net core but released nuget is old.

Could you please help me with another library, Microsoft.Azure.ServiceBus.EventProcessorHost ? ServiceFabric is already able to use with .net core but this key library is still not. Do you have any info regarding this please?

thanks a lot

**Comment by tameraw** [comment]
_From @olivierbloch on October 27, 2016 14:46_

Hi @ondrejtomcik 
I'll work on getting the Nuget Package updated for the azure-amqp library.
For the EventProcessorHost one, it's a different team owning it, I'll try and see what their plans are. If we could get the C2D messaging and the registry manager working already that'd be great.

**Comment by tameraw** [comment]
_From @ondrejtomcik on October 27, 2016 19:9_

@olivierbloch great :)
One more question, when will be twins functionality released? I should maybe start to migrate from azure-iot-sdks-preview, no?

Let me know if you have some info regarding EventProcessorHost please.
Thanks

**Comment by tameraw** [comment]
_From @olivierbloch on October 27, 2016 19:41_

@ondrejtomcik 
We expect to have the preview repo merged back into the mainstream one next month.
But I recommend you work in the current repo for this work.

**Comment by tameraw** [comment]
_From @olivierbloch on October 27, 2016 19:51_

@ondrejtomcik 
Regarding EventProcessorHost support for .Net Core, it seems there is work in progress, but not ETA yet.
In the meantime we could consider using https://github.com/Azure/amqpnetlite

**Comment by tameraw** [comment]
_From @ondrejtomcik on November 10, 2016 19:39_

@olivierbloch 
How it looks with the azure amqp nuget release ?

**Comment by tameraw** [comment]
@ondrejtomcik - ETA before end of this month.

**Comment by tameraw** [comment]
_From @radu-matei on December 2, 2016 13:45_

Any updates on the .NET Core support for the IoT Hub SDK?

Is there any way I can get to help?

Thanks!

**Comment by tameraw** [comment]
_From @ondrejtomcik on December 5, 2016 9:0_

Hi Radu,
I was waiting for amqp release, but in a meantime we switched to different IoT platform. IoT Hub does not meet our criteria after deep evaluation.

**Comment by tameraw** [comment]
_From @Kordonme on December 13, 2016 21:46_

+1 Bummer to see that Microsoft.Azure.Devices is not .NET Core compatible. Updated ETA, or should we get our hands dirty?

**Comment by yfakariya** [comment]
Is there any roadmap or status of this? Waiting #35 fix?
I'm waiting .NET Core build to use C# device SDK's features in linux gateways.

I also have private fork for .NET Standard and ready to PR if you will accept.

**Comment by darrenhull** [comment]
The event hubs library for .net  Standard has been released, do we have an update on the devices library port? Does #35 still need fixing? I'm able to help with some direction.

**Comment by darrenhull** [comment]
@tameraw the comment above "I suggest creating a shared project as shown here" 404's, can you provide another example?

**Comment by darrenhull** [comment]
@tameraw found the link: https://github.com/Azure/azure-iot-sdk-csharp/tree/master/shared

**Comment by adminnz** [comment]
Whats the status of this?
IoT Hub is missing its target if it isnt even going to work on the majority of IoT Devices (low powered linux ARM & Atom platforms)

**Comment by yfakariya** [comment]
The release note says that some works are in progress, but it is not clear. What items are still working?

**Comment by yfakariya** [comment]
...and I found that latest NuGet package does not contain .NET Standard dll.  
https://www.nuget.org/packages/Microsoft.Azure.Devices.Client

Is it intended?

**Comment by WillooWisp** [comment]
I don't see any .NET Core related targets in the NuGet spec, but .NET Standard is mentioned in the docs. What's the .NET Core status? MQTT is not working for example, and using AMQP does not work either, since it complains about missing DLL. {System.IO.FileNotFoundException: Could not load file or assembly 'Microsoft.Azure.Amqp, Version=2.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35'. The system cannot find the file specified.
File name: 'Microsoft.Azure.Amqp, Version=2.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35'

**Comment by yfakariya** [comment]
Hi @tameraw, any information about our questions? I checked latest (v1.2.7) nupkg and saw that it did not contain netstandard build. Thanks!

**Comment by darrenhull** [comment]
Do we have an update on this? Is there a preview or beta we could try?

**Comment by ancutapasolea** [comment]
Hello, any updates about .Net Core support?

**Comment by frnk01** [comment]
Updates please. Need .netcore support.

**Comment by olivierbloch** [comment]
.Net Core support has been released to the Device client SDK (Microsoft.Azure.Devices.Client 1.2.9)
Go ahead and get the latest [NuGet Package ](https://www.nuget.org/packages/Microsoft.Azure.Devices.Client/) form NuGet to try it out.
We are now working on the Service client an should have it three too soon.

**Comment by tameraw** [comment]
.Net Core support now available for Service client as well.

**Comment by Jose-CF** [comment]
Which version is currently supported? I am trying to run .NET Core 2.0 application in service fabric and seems unsupported

**Comment by oycbox** [comment]
Microsoft.Azure.Devices.Client 1.6.2 on .NET Core 2.0 console application still not supported? i got silent error out when running the Device sample code on dotnet core.

**Comment by CIPop** [comment]
@oycbox, we support NetStandard1.3 for a while and the console app should work. I'm currently working on a larger cleanup and port to netstandard2.0, available in the next release.

Please create a new issue and describe the bug. Please include a simple code example (or point at the sample that doesn't work).

**Comment by androidand** [comment]
Any news for .NET Core 2.0 and Azure IoT? Would be awesome to se an example with a .NET Core API app.

**Comment by CIPop** [comment]
@androidand Yes! We now have full .Net Standard 2.0 starting with the [3-18-2018 release](https://github.com/Azure/azure-iot-sdk-csharp/releases/tag/2018-3-13)!
This was a complete overhaul of the build system, conditional compilation, etc. Please let us know what you think!

Please open separate issues if you find bugs.
